spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/ConfigurationPropertyValueHintRecoveryStrategy.java

```diff
@@ -85,7 +86,7 @@ public void addProposals(String dsl, CheckPointedParseException exception, int d
 
 					File appFile = appResource.getFile();
 					Archive jarFileArchive = appFile.isDirectory() ? new ExplodedArchive(appFile) : new JarFileArchive(appFile);
-					classLoader = new ClassLoaderExposingJarLauncher(jarFileArchive).createClassLoader();
+					classLoader = new BootClassLoaderCreation(jarFileArchive, Thread.currentThread().getContextClassLoader()).createClassLoader();
```

Comment:
Do we really need to set the entire current context loader as `parent` here? I thought the URLs derived from the `jarFileArchive` are sufficient to create the class-loader and to load the metadata files.